dbt/models/qc/qc.vw_report_town_close_res_edit.sql

```diff
@@ -0,0 +1,193 @@
+    FROM (
+        SELECT
+            parid,
+            taxyr,
+            MAP_AGG(row_num, yrblt) AS yrblt_kv,
+            MAP_AGG(row_num, sfla) AS sfla_kv,
+            MAP_AGG(row_num, stories) AS stories_kv,
+            MAP_AGG(row_num, rmbed) AS rmbed_kv,
+            MAP_AGG(row_num, fixbath) AS fixbath_kv,
+            MAP_AGG(row_num, fixhalf) AS fixhalf_kv,
+            MAP_AGG(row_num, bsmt) AS bsmt_kv,
+            MAP_AGG(row_num, bsmt_fin) AS bsmt_fin_kv,
+            MAP_AGG(row_num, external_propct) AS external_propct_kv,
+            MAP_AGG(row_num, alt_cdu) AS alt_cdu_kv
+        FROM first_five_dwellings
+        GROUP BY parid, taxyr
+    ) AS agg_dwellings
+),
+
+num_dwellings AS (
+    SELECT
+        parid,
+        taxyr,
+        COUNT(*) AS num_dwellings
+    FROM first_five_dwellings
+    GROUP BY parid, taxyr
+)
+
+SELECT
+    pardat.parid,
+    pardat.taxyr,
+    legdat.user1 AS township_code,
+    pardat.nbhd,
+    pardat.class,
+    legdat.adrno,
+    legdat.adradd,
+    legdat.adrdir,
+    legdat.adrstr,
+    legdat.adrsuf,
+    legdat.adrsuf2,
+    legdat.unitdesc,
+    legdat.unitno,
+    legdat.cityname,
+    aprval.aprland,
+    aprval.aprbldg,
+    aprval.aprtot,
+    aprval_prev.aprland AS aprland_prev,
+    aprval_prev.aprbldg AS aprbldg_prev,
+    aprval_prev.aprtot AS aprtot_prev,
+    CONCAT(
+        CAST(
+            ROUND(
+                (
+                    (aprval.aprtot - aprval_prev.aprtot)
+                    / CAST(aprval_prev.aprtot AS DOUBLE)
+                )
+                * 100,
+                2
+            )
+            AS VARCHAR
+        ),
+        '%'
+    ) AS aprtot_percent_change,
+    aprval.dwelval,
+    aprval.dwelval + aprval.aprland AS dweltot,
+    aprval_prev.dwelval AS dwelval_prev,
+    aprval_prev.dwelval + aprval_prev.aprland AS dweltot_prev,
+    CONCAT(
+        CAST(
+            ROUND(
+                (
+                    (
+                        (aprval.dwelval + aprval.aprland)
+                        - (aprval_prev.dwelval + aprval_prev.aprland)
+                    )
+                    / CAST(
+                        (aprval_prev.dwelval + aprval_prev.aprland) AS DOUBLE
+                    )
+                )
+                * 100,
+                2
+            )
+            AS VARCHAR
+        ),
+        '%'
+    ) AS dweltot_percent_change,
+    sale.saledt_fmt,
+    sale.price,
+    sale.instruno,
+    COALESCE(pardat.tiebldgpct, 0) AS tiebldgpct,
+    aprval.reascd,
+    aprval.obyval,
+    land.sf,
+    num_dwellings.num_dwellings,
+{% for idx in range(1, 6) %}
+    dweldat.yrblt_{{ idx }},
+    dweldat.sfla_{{ idx }},
+    dweldat.stories_{{ idx }},
+    dweldat.rmbed_{{ idx }},
+    dweldat.fixbath_{{ idx }},
+    dweldat.fixhalf_{{ idx }},
+    dweldat.bsmt_{{ idx }},
+    dweldat.bsmt_fin_{{ idx }},
+    dweldat.external_propct_{{ idx }},
+    dweldat.alt_cdu_{{ idx }}{% if not loop.last %},{% endif %}
+{% endfor %}
```

Comment:
It would be nice to recode these values a la `ccao::vars_recode()`, is there a way to do that in Athena? Or perhaps in Python as part of the export script?